src/Context/GlobalContext.js

```diff
@@ -0,0 +1,41 @@
+import { createContext, useContext, useState } from 'react';
+
+export const ShopCartContext = createContext({
+    myCart: {},
+    addToMyCart: () => {},
+    removeFromMyList: () => {},
+});
+
+export const ShopCartState = ({children}) =>{
+    const [myCart, setMyCart] = useState({});
+
+    const addToMyCart = (productId, productdata, qty = 1) => {
+        
+        setMyCart( prevState => ({
+            ...prevState,
+            [productId]: {...productdata, qty}
+        }));
+    };
+
+    const removeFromMyList = (product) => {
+        if(!(product.id in myCart)) return
+
+        setMyCart( prevState => {
+            const newState = {...prevState}
+            delete newState[product.id]
+            return newState
+        } )
+    }
```

Comment:
Not sure if you already saw the `useReducer` hook for context management. But we can sync up about it later in the afternoon. Check: https://reactjs.org/docs/hooks-reference.html#usereducer